### ISSUE TITLE
chunkwise--triton_xl_chunk backward returns incorrect gradients on Blackwell (sm_120) / Windows; forward is correct, training loss stalls

### ISSUE THREAD

**Issue by Heatdh** [NX-AI/mlstm_kernels#16]
## Summary

On an RTX 5080 (Blackwell, `sm_120`) under Windows, the `chunkwise--triton_xl_chunk`
kernel produces a **correct forward pass but incorrect gradients in the backward
pass**. The gradients are *finite* (no NaN/Inf), but they disagree with the
`chunkwise--native_autograd` reference by ~100% for `dq`, `dk`, `di`, `df` (only `dv`
matches). As a result, training with the Triton kernel does not learn — the loss
stalls — while the native kernel on the identical setup trains normally.

## Environment

- GPU: NVIDIA GeForce RTX 5080 (Blackwell, compute capability `12.0` / `sm_120`), 16 GB
- OS: Windows 11
- Python 3.13
- `torch` 2.11.0+cu128
- `triton-windows` 3.7.1.post27
- `mlstm_kernels` 2.0.2
- `xlstm` 2.0.5

## Minimal reproduction

Compares `chunkwise--native_autograd` (reference autograd) against
`chunkwise--triton_xl_chunk` on identical inputs, in float32, deterministic seed:

```python
import torch
from mlstm_kernels.torch.backend_module import mLSTMBackend, mLSTMBackendConfig

dev = "cuda"
B, NH, S, DHQK, DHHV = 1, 4, 256, 64, 64
print("torch", torch.__version__, "| cap", torch.cuda.get_device_capability(0))


def make_inputs():
    g = torch.Generator(device=dev).manual_seed(1234)
    q = torch.randn(B, NH, S, DHQK, device=dev, generator=g, requires_grad=True)
    k = torch.randn(B, NH, S, DHQK, device=dev, generator=g, requires_grad=True)
    v = torch.randn(B, NH, S, DHHV, device=dev, generator=g, requires_grad=True)
    i = torch.randn(B, NH, S, device=dev, generator=g, requires_grad=True)
    f = torch.randn(B, NH, S, device=dev, generator=g, requires_grad=True)
    return q, k, v, i, f


def run(kernel):
    cfg = mLSTMBackendConfig(chunkwise_kernel=kernel, mode="train",
                             return_last_states=False, chunk_size=64,
                             autocast_kernel_dtype="float32")
    be = mLSTMBackend(cfg).to(dev)
    q, k, v, i, f = make_inputs()
    out = be(q=q, k=k, v=v, i=i, f=f)
    out = out[0] if isinstance(out, tuple) else out
    out.float().pow(2).mean().backward()
    return out.detach().float(), {n: t.grad for n, t in zip("qkvif", (q, k, v, i, f))}

on, gn = run("chunkwise--native_autograd")
ot, gt = run("chunkwise--triton_xl_chunk")
print("forward max|native-triton| =", (on - ot).abs().max().item())
for n in "qkvif":
    d = (gn[n].float() - gt[n].float()).abs().max().item()
    rel = d / (gn[n].float().abs().max().item() + 1e-8)
    print(f"grad d{n}: max abs diff = {d:.3e}  (rel {rel:.3e})  "
          f"triton_finite={torch.isfinite(gt[n]).all().item()}")
```

### Output on the machine above

```
torch 2.11.0+cu128 | cap (12, 0)
forward max|native-triton| = 5.036e-02
grad dq: max abs diff = 1.589e-01  (rel 9.825e-01)  triton_finite=True
grad dk: max abs diff = 1.468e-01  (rel 9.188e-01)  triton_finite=True
grad dv: max abs diff = 1.478e-05  (rel 5.298e-03)  triton_finite=True
grad di: max abs diff = 3.912e-01  (rel 1.184e+00)  triton_finite=True
grad df: max abs diff = 1.162e-01  (rel 1.174e+00)  triton_finite=True
```

`dv` matches to ~5e-3; `dq`, `dk`, `di`, `df` are effectively uncorrelated with the
reference (relative error ~1). The forward output agrees to ~5e-2 (loss within 0.2%).

## Downstream effect

Training a small xLSTM LM with `chunkwise--triton_xl_chunk` leaves cross-entropy
essentially flat (7.8507 → 7.8525 over 120 steps), while `chunkwise--native_autograd`
on the same data/model descends normally (7.77 → 7.37). Forward throughput with the
Triton kernel is ~2.45× native (≈245k vs ≈100k tok/s), so the kernel is otherwise
working — the issue is confined to the backward gradients.

## Notes / hypotheses

- Forward correctness with a broken backward points at the custom backward kernels
  (`bw_kernel_parallel_dQ/dK`, `bw_recurrent`, gate-gradient accumulation), not the
  recurrence math.
- Could be `sm_120`-specific codegen in `triton-windows` rather than the kernel
  source. I have not yet been able to test on Linux / an older arch to isolate
  Windows-vs-Blackwell — happy to if that would help bisect.

## Offer

I can provide per-parameter grad tensors, sweep head dims / chunk sizes / dtypes, test
`triton_xl_chunk_siging` and `triton_limit_chunk`, or try a specific commit. Let me
know what would be most useful.
